packages/sync/src/class-functions.php

```diff
@@ -161,22 +161,83 @@ public static function get_post_type_features() {
 	 * @return string Hosting provider.
 	 */
 	public static function get_hosting_provider() {
-		if ( defined( 'GD_SYSTEM_PLUGIN_DIR' ) || class_exists( '\\WPaaS\\Plugin' ) ) {
-			return 'gd-managed-wp';
-		}
-		if ( defined( 'MM_BASE_DIR' ) ) {
-			return 'bh';
+		$hosting_provider_detection_methods = array(
+			'get_hosting_provider_by_known_constant',
+			'get_hosting_provider_by_known_class',
+			'get_hosting_provider_by_known_function',
+		);
+
+		$functions = new Functions();
+		foreach ( $hosting_provider_detection_methods as $method ) {
+			$hosting_provider = call_user_func( array( $functions, $method ) );
+			if ( false !== $hosting_provider ) {
+				return $hosting_provider;
+			}
 		}
-		if ( defined( 'IS_PRESSABLE' ) ) {
-			return 'pressable';
+
+		return 'unknown';
+	}
+
+	/**
+	 * Return a hosting provider using a set of known constants.
+	 *
+	 * @return mixed A host identifier string or false.
+	 */
+	public function get_hosting_provider_by_known_constant() {
+		$hosting_provider_constants = array(
+			'GD_SYSTEM_PLUGIN_DIR' => 'gd-managed-wp',
+			'MM_BASE_DIR'          => 'bh',
+			'PAGELYBIN'            => 'pagely',
+			'KINSTAMU_VERSION'     => 'kinsta',
+			'FLYWHEEL_CONFIG_DIR'  => 'flywheel',
+			'IS_PRESSABLE'         => 'pressable',
+			'VIP_GO_ENV'           => 'vip-go',
+		);
+
+		foreach ( $hosting_provider_constants as $constant => $constant_value ) {
+			if ( Constants::is_defined( $constant ) ) {
+				if ( 'VIP_GO_ENV' === $constant && false === Constants::get_constant( 'VIP_GO_ENV' ) ) {
```

Comment:
I wonder if it makes sense to apply this `false` check to all of the constants? Maybe something like:

```
if ( Constants::is_defined( $constant ) && false !== Constants::get_constant( 'VIP_GO_ENV' )  ) {
  return $constant_value;
}
```

This isn't a blocker, but it could minimize the need for us to change this in the future.

Reply:
Wasn't sure on this as this logic is from before the update and specific to vip-go but it's a good call. I've updated this for everything.

Reply:
Seems like this still needs to be updated though as it's `VIP_GO_ENV`-specific?